bundles/org.openhab.binding.ferroamp/src/main/java/org/openhab/binding/ferroamp/internal/FerroampHandler.java

```diff
@@ -0,0 +1,260 @@
+    private static List<FerroampChannelConfiguration> channelConfigSsoS4 = new ArrayList<>();
+    private static List<FerroampChannelConfiguration> channelConfigEso = new ArrayList<>();
+    private static List<FerroampChannelConfiguration> channelConfigEsm = new ArrayList<>();
+
+    public FerroampHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        String transId = UUID.randomUUID().toString();
+        String valueConfiguration = command.toString();
+        if (FerroampBindingConstants.CHANNEL_REQUESTCHARGE.equals(channelUID.getId())) {
+            String requestCmdJsonCharge = "{\"" + "transId" + "\":\"" + transId
+                    + "\",\"cmd\":{\"name\":\"charge\",\"arg\":\"" + valueConfiguration + "\"}}";
+            FerroampMqttCommunication.sendPublishedTopic(requestCmdJsonCharge, ferroampConfig);
+        }
+        if (FerroampBindingConstants.CHANNEL_REQUESTDISCHARGE.equals(channelUID.getId())) {
+            String requestCmdJsonDisCharge = "{\"" + "transId" + "\":\"" + transId
+                    + "\",\"cmd\":{\"name\":\"discharge\",\"arg\":\"" + valueConfiguration + "\"}}";
+            FerroampMqttCommunication.sendPublishedTopic(requestCmdJsonDisCharge, ferroampConfig);
+        }
+        if (FerroampBindingConstants.CHANNEL_AUTO.equals(channelUID.getId())) {
+            String requestCmdJsonAuto = "{\"" + "transId" + "\":\"" + transId + "\",\"cmd\":{\"name\":\"auto\"}}";
+            FerroampMqttCommunication.sendPublishedTopic(requestCmdJsonAuto, ferroampConfig);
+        }
+    }
+
+    @Override
+    public void initialize() {
+        // Set configuration parameters
+        ferroampConfig = getConfigAs(FerroampConfiguration.class);
+
+        // Set channel configuration parameters
+        channelConfigEhub = FerroampChannelConfiguration.getChannelConfigurationEhub();
+        channelConfigSsoS1 = FerroampChannelConfiguration.getChannelConfigurationSsoS1();
+        channelConfigSsoS2 = FerroampChannelConfiguration.getChannelConfigurationSsoS2();
+        channelConfigSsoS3 = FerroampChannelConfiguration.getChannelConfigurationSsoS3();
+        channelConfigSsoS4 = FerroampChannelConfiguration.getChannelConfigurationSsoS4();
+        channelConfigEso = FerroampChannelConfiguration.getChannelConfigurationEso();
+        channelConfigEsm = FerroampChannelConfiguration.getChannelConfigurationEsm();
+
+        if (!ferroampConfig.hostName.isBlank() || !ferroampConfig.password.isBlank()
+                || !ferroampConfig.userName.isBlank()) {
+            final MqttBrokerConnection ferroampConnection = new MqttBrokerConnection(ferroampConfig.hostName,
+                    FerroampBindingConstants.BROKER_PORT, false, false, ferroampConfig.userName);
+            updateStatus(ThingStatus.UNKNOWN);
+            scheduler.scheduleWithFixedDelay(this::pollTask, 60, ferroampConfig.refreshInterval, TimeUnit.SECONDS);
```

Comment:
This class also needs a dispose method that clears up any resources like active connections and scheduled tasks.

Reply:
Has added the dispose method.